certigo connect dereferences nil when using Postgres StartTLS

When trying to use `certigo connect -t postgres`, the following happens:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13d95d3]
goroutine 1 [running]:
github.com/square/certigo/starttls.GetConnectionState(0x7ffeefbff963, 0x4, 0x0, 0x0, 0x7ffeefbff968, 0xe, 0x150977e, 0x7, 0x0, 0x0, ...)
	/private/tmp/certigo-20200303-20580-1o9foqt/certigo-1.12.1/starttls/starttls.go:258 +0x3b3
github.com/square/certigo/cli.Run(0xc0000200b0, 0x4, 0x4, 0x15c2ce0, 0xc0000101e8, 0x0)
	/private/tmp/certigo-20200303-20580-1o9foqt/certigo-1.12.1/cli/cli.go:131 +0xbb4
main.main()
	/private/tmp/certigo-20200303-20580-1o9foqt/certigo-1.12.1/main.go:27 +0xae
```
This is because the final return statement of `starttls.GetConnectionState` [dereferences `cri`][deref] (a double-pointer), which unfortunately [is never initialized when using Postgres StartTLS][init]. In theory this could also happen above [during the check for old versions of TLS][oldtls], but I haven't tested that codepath.

[deref]: https://github.com/square/certigo/blob/v1.12.1/starttls/starttls.go#L258
[init]: https://github.com/square/certigo/blob/v1.12.1/starttls/starttls.go#L132-L140
[oldtls]: https://github.com/square/certigo/blob/v1.12.1/starttls/starttls.go#L250

--- a/starttls/starttls.go
+++ b/starttls/starttls.go
@@ -247,6 +247,10 @@ func GetConnectionState(startTLSType, connectName, connectTo, identity, clientCe
 		return nil, nil, fmt.Errorf("error connecting: %v", result.err)
 	}
 
+	if cri == nil {
+		return result.state, nil, nil
+	}
+
 	if result.state.Version < tls.VersionTLS12 && *cri != nil {
 		// Sending supported signature schemes was added in TLS 1.2,
 		// but Go lies to us in the GetClientCertificate callback and

